asr1k_neutron_l3/plugins/l3/agents/asr1k_l3_agent.py

```diff
@@ -724,9 +733,14 @@ def _process_routers_loop(self):
             LOG.warning("The processing thread pool size has been reduced to match 'yang_connection_pool_size' "
                         "its now {}".format(poolsize))
 
-        pool = eventlet.GreenPool(size=poolsize)
-        while True:
-            pool.spawn_n(self._process_router_update)
+        with ThreadPoolExecutor(max_workers=poolsize) as executer:
```

Comment:
Creating a thread pool every N minutes is much more resource-intensive than creating green threads. Perhaps we should think about creating a thread pool and reusing it.

Reply:
Where do you see that a thread pool is created every N minutes, and where do you take these N minutes?

This threadpool is created exactly once. And this doesn't even mean that the threads of the pool are created - they are spun up as demand rises (and jobs submitted to the executer).

Reply:
I did not understand the logic below. Question resolved.